scripts/o3de/o3de/repo.py

```diff
@@ -79,6 +70,84 @@ def process_add_o3de_repo(file_name: str or pathlib.Path,
             download_file_result = utils.download_file(parsed_uri, cache_file, True)
             if download_file_result != 0:
                 return download_file_result
+    return 0
+
+def validate_remote_repo(repo_uri: str, validate_contained_objects: bool = False) -> bool:
+    manifest_uri = get_repo_manifest_uri(repo_uri)
+
+    if not manifest_uri:
+        return False
+
+    cache_file = download_repo_manifest(manifest_uri)
+
+    if not cache_file:
+        return False
+
+    if not validation.valid_o3de_repo_json(cache_file):
+        return False
+
+    if validate_contained_objects:
+        repo_data = {}
+        with file_name.open('r') as f:
+            try:
+                repo_data = json.load(f)
+            except json.JSONDecodeError as e:
+                return False
+
+        download_object_manifest_result = download_object_manifests(repo_data)
+        if download_object_manifest_result != 0:
+            return False
```

Comment:
don't think you're using the temporary local var
```suggestion
        if download_object_manifests(repo_data) = 0:
            return False
```